libcommon/strjoin.c

```diff
@@ -0,0 +1,39 @@
+/*
+ * strjoin.c - concatenates strings from an array into a single one
+ *
+ * Mildly inspired by Go standard library's strings.Join function.
+ *
+ * Copyright (c) 2025 Luiz Antônio Rangel (takusuman)
+ *
+ * SPDX-Licence-Identifier: Zlib
+ */
+
+#include <stdlib.h>
+#include <string.h>
+
+char *strjoin(const char *elems[], char *sep) {
+ 	size_t en = 0,
+	       i = 0,
+	       joinlen = 0;
+	const char **elem = NULL;
+	char *joinbuf = NULL,
+	     *joinbufp = NULL;
+
+	for (elem = elems; *elem; elem++, en++)
+		joinlen += strlen(*elem);
+	joinlen += ((strlen(sep) * (en - 1)) + 1);
+
+	/* Populate the buffer. */
+	joinbuf = malloc((joinlen * sizeof(char)));
+	joinbufp = joinbuf;
+
+	for (i = 0; i < en; i++) {
+		stpcpy((joinbufp + strlen(joinbufp)),
```

Comment:
Why are you doing this shit? The point of `stpcpy` instead of `strcpy` is not having to use `strlen`

Reply:
So it's the exactly same as using strcat. Why the fuck would I ask you to change it to something that does the exactly same? How is it any different in efficiency? Can't you just read the code I sent?